classes/stageAssignment/StageAssignment.php

```diff
@@ -90,7 +90,7 @@ public function scopeWithStageIds(Builder $query, ?array $stageIds): Builder
     public function scopeWithSubmissionIds(Builder $query, ?array $submissionIds): Builder
     {
         return $query->when($submissionIds !== null, function ($query) use ($submissionIds) {
-            return $query->whereIn('submission_id', $submissionIds);
+            return $query->whereIn('stage_assignments.submission_id', $submissionIds);
```

Comment:
When using the `scopeWithSubmissionIds` filter in combination with another filter, such as `scopeWithContextId`, which performs joins with tables containing a column named `submission_id`, you'll encounter an ambiguous column error because the column name `submission_id` exists in multiple tables in the query.

So for example the following query would throw the error below
```
$stageAssignments = StageAssignment::withSubmissionIds([$submission->getId()])
     ->withStageIds($stageId ? [$stageId] : [])
     ->withUserId($user->getId())
     ->withContextId($this->context->getId()) // joins submissions table which also has a submission_id column
     ->get();
````
**Error**
```
SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'submission_id' in where clause is ambiguous (Connection: mysql, SQL: select * from `stage_assignments` inner join `submissions` on `stage_assignments`.`submission_id` = `submissions`.`submission_id` where `submission_id` in (9) and exists (select * from `user_group_stage` where `stage_assignments`.`user_group_id` = `user_group_stage`.`user_group_id` and `stage_id` in (5)) and `user_id` = 3 and `submissions`.`context_id` = 1)```

Reply:
Good to know!